### PR TITLE
GitHub Action to lint Python code

### PR DESCRIPTION
**Pull request by cclauss**
Because Travis CI is no longer running on PRs.
Output: https://github.com/cclauss/gspread/actions

### CODE REVIEW

**Comment by lavigne958** [comment]
Thank you @cclauss  for this contribution, this is great, I just though that we need some kind of linting in the project.

Though you use `<command> || true` in all `run` steps here so if something fails it passes anyway. :thinking:  any reason for this ?

As strange as it could be, I don't know enough python to know all the tools you setup, I know black is good for code format.

I would need help form someone else to check if these tools are necessary and correctly setup :slightly_smiling_face:

**Comment by cclauss** [comment]
The idea is to land this PR so that the tests are running and then do future PRs that remove the `|| true` from various tests as they pass or remove the tests that are deemed to be unnecessary.

**Comment by lavigne958** [comment]
That's a good way to proceed, but I can't take full responsibility on this, @burnash what do you think ?

**Comment by cclauss** [comment]
$ `pytest --doctest-modules .`
```
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: /home/runner/work/gspread/gspread
plugins: betamax-0.8.1
collected 18 items

gspread/client.py FFFF                                                   [ 22%]
gspread/models.py FFFFF                                                  [ 50%]
gspread/utils.py .......FF                                               [100%]
```

**Comment by lavigne958** [comment]
> The idea is to land this PR so that the tests are running and then do future PRs that remove the `|| true` from various tests as they pass or remove the tests that are deemed to be unnecessary.

Hi @cclauss I have not forget about your PR, I'm still thinking about how to introduce it. 
So far I think I will make it land in the repository for version 4.0.0 comming in some time sonn.

File: .github/workflows/lint_python.yml
Comment (lavigne958) [inline .github/workflows/lint_python.yml:17]: I ran some scenarios, I found a way to run the tests in the CI, I will need to update the files under `tests/cassettes/` then this should pass. :rocket: 

*Important:* we should not provide a value in `GS_CREDS_FILENAME` if we wish to run i in the CI, one should only do so to run the tests using a real HTTP client and make each requests. :teacher: 

I'll advise to remove that comment, ro prevent confusion :slightly_smiling_face:

File: .github/workflows/lint_python.yml
Comment (lavigne958) [inline .github/workflows/lint_python.yml:19]: This should be removed too, now one can run the tests using `make test` which will run tox which will do the setup for each available version of python.

File: .github/workflows/lint_python.yml
Comment (lavigne958) [inline .github/workflows/lint_python.yml:15]: we should only need `tox`, then the reste will be downloaded/setup when starting the test.

**Review by lavigne958 — COMMENTED** [review]
Thanks again @cclauss  for that PR.

I will merge it once fixed.

It will be a good start for next release 5.0.0

**Comment by cclauss** [comment]
Please re-review.  Thanks

File: .github/workflows/lint_python.yml
Comment (cclauss) [reply .github/workflows/lint_python.yml]: Removed

File: .github/workflows/lint_python.yml
Comment (cclauss) [reply .github/workflows/lint_python.yml]: GitHub Actions can take secrets but removed anyways.

File: .github/workflows/lint_python.yml
Comment (cclauss) [reply .github/workflows/lint_python.yml]: Removed.

File: .github/workflows/lint_python.yml
Comment (cclauss) [reply .github/workflows/lint_python.yml]: Removed.

File: .github/workflows/lint_python.yml
Comment (lavigne958) [reply .github/workflows/lint_python.yml]: it is not a matter of storing secrets it's the way the test suite is configured. if credentials are provided then it will try to connect to the internet and make the real request, if not provided and the default record mode is set to `once` or `none`then no actual connection to a real spreadsheet is made.

File: HISTORY.rst
Comment (lavigne958) [inline HISTORY.rst:13]: googd catch ! :smile:

File: .github/workflows/lint_python.yml
Comment (cclauss) [reply .github/workflows/lint_python.yml]: Removed anyways.

**Review by lavigne958 — APPROVED** [review]
LGTM. I will merge your PR once I made the release 4.0.1

File: HISTORY.rst
Comment (cclauss) [reply HISTORY.rst:13]: Codespell caught it.

**Comment by cclauss** [comment]
If we are going to land this one then we should do so as bandit, codespell, flake8, and safety will flag issues in future pull requests.  After it lands, we can tighten up other tests.

**Comment by lavigne958** [comment]
I'm not too sure to fully understand what you mean.
You mean we should land this PR then fix everything so bandit can pass, then fix everything so codespell can pass and so on?

If so I am OK with that.

**Comment by cclauss** [comment]
bandit, codespell, flake8, and safety are already active and pass.  It will be the other tests that we will need to fix after this PR lands.

**Comment by cclauss** [comment]
I have done what I could.  Both nose and pytests fail but this PR should be merged and then fixed.

**Comment by lavigne958** [comment]
Yep don't worry once it is setup I'll have a look and I'll work on it.

Thanks again for your work and patiente.

**Review by lavigne958 — APPROVED** [review]

**Pull request merged**
